### PR TITLE
feat: move streamr to cache-service

### PR DESCRIPTION
**Pull request by koonopek**
* refactor of broadcasting to some concept as in node
* move streamr brodcaster to gateway
* we have to generate single private key for streamr and send there matic

### CODE REVIEW

File: packages/cache-service/src/app.module.ts [hunk 1]
```diff
@@ -4,25 +4,33 @@ import { ScheduleModule } from "@nestjs/schedule";
...
 import { StreamrListenerService } from "./streamr-listener/streamr-listener.service";
+import { StreamrBroadcaster } from "./brodcasters/streamr-brodcaster";
+import { BundlrBroadcaster } from "./brodcasters/bundlr-brodcaster";
+import { MongoBroadcaster } from "./brodcasters/mongo-broadcaster";
 
-const providers: Provider[] = [DataPackagesService, BundlrService];
+const providers: Provider[] = [
+  DataPackagesService,
+  BundlrBroadcaster,
+  MongoBroadcaster,
```
Comment (koonopek) [inline packages/cache-service/src/app.module.ts:20]: It took my 4 hours to find that I was missing this line :( cry with me

File: packages/cache-service/src/brodcasters/data-pacakges-brodcaster.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { CachedDataPackage } from "../data-packages/data-packages.model";
```
Comment (cehali) [inline packages/cache-service/src/brodcasters/data-pacakges-brodcaster.ts:1]: Typo in filename `data-pacakges`

File: packages/cache-service/src/data-packages/data-packages.service.ts [hunk 3]
```diff
@@ -23,51 +35,87 @@ import {
...
 
-  // eslint-disable-next-line @typescript-eslint/class-methods-use-this
-  async saveManyDataPackagesInDB(dataPackages: CachedDataPackage[]) {
-    await DataPackage.insertMany(dataPackages);
+  private performBroadcast(
+    dataPackagesToSave: CachedDataPackage[],
+    nodeEvmAddress: string,
+    { instance: broadcaster, name }: BroadcasterInfo
+  ): Promise<void> {
+    const message = `broadcast ${dataPackagesToSave.length} data packages for node ${nodeEvmAddress}`;
```
Comment (cehali) [inline packages/cache-service/src/data-packages/data-packages.service.ts:103]: Maybe add to the message which broadcaster succeeded/failed

File: packages/cache-service/test/e2e/data-packages.spec.ts [hunk 4]
```diff
@@ -84,6 +83,9 @@ describe("Data packages (e2e)", () => {
   let mockDataPackages: SignedDataPackagePlainObj[];
 
   beforeEach(async () => {
+    // Connect to mongoDB in memory
+    await createTestDB();
```
Comment (cehali) [inline packages/cache-service/test/e2e/data-packages.spec.ts:87]: [Minor] WDYT renaming to `createAndConnectToTestDb`, to indicate that after this function you are connected to db

**Review by cehali — DISMISSED** [review]
GJ 👍  two minor comments

File: yarn.lock [hunk 5]
```diff
@@ -7265,7 +7278,7 @@ axios@^0.27.0, axios@^0.27.2:
     follow-redirects "^1.14.9"
     form-data "^4.0.0"
 
-axios@^1.0.0, axios@^1.5.0:
```
Comment (rpiszczatowski) [inline yarn.lock:7268]: It looks like we changed it one way and then back. Our code AFAIR uses axios 1.5.1 at the moment.

File: yarn.lock [hunk 6]
```diff
@@ -5304,19 +5304,32 @@
...
+"@ton-community/func-js@0.6.3-tvmbeta.1":
   version "0.6.3-tvmbeta.1"
   resolved "https://registry.yarnpkg.com/@ton-community/func-js/-/func-js-0.6.3-tvmbeta.1.tgz#90ec7987f341d1a56862a72b6b0415eb42c04bbb"
   integrity sha512-6/Iz3bbDQFMzyFUc0tOLukY2l+ItXZJt7q5Jcs8edHlBsYhffHUIgXinPE7E70Bb0E3Cbs3p+8RxPSLLrUDGhQ==
   dependencies:
     "@ton-community/func-js-bin" "0.4.5-tvmbeta.1"
     arg "^5.0.2"
 
+"@ton-community/func-js@^0.6.2":
+  version "0.6.2"
```
Comment (rpiszczatowski) [inline yarn.lock:5326]: I don't think this change is needed. Is it?

File: packages/cache-service/test/e2e/data-packages.spec.ts [hunk 7]
```diff
@@ -93,14 +95,11 @@ describe("Data packages (e2e)", () => {
     await app.init();
     httpServer = app.getHttpServer();
 
-    const bundlrService = app.get(BundlrService);
-    bundlrSaveDataPackagesSpy = jest.spyOn(bundlrService, "saveDataPackages");
+    const bundlrService = app.get(BundlrBroadcaster);
```
Comment (rpiszczatowski) [inline packages/cache-service/test/e2e/data-packages.spec.ts:98]: bundlrService -> bundlerBroadcaster?

**Review by rpiszczatowski — COMMENTED** [review]
Some minor notes.

File: yarn.lock (hunk 6 above)
Comment (koonopek) [reply yarn.lock:5326]: It is spawnning randomly when I install anything with yarn,  should I edit manually yarn.lock every time?

File: packages/cache-service/test/e2e/data-packages.spec.ts (hunk 4 above)
Comment (koonopek) [reply packages/cache-service/test/e2e/data-packages.spec.ts:87]: good idea

File: yarn.lock (hunk 5 above)
Comment (koonopek) [reply yarn.lock:7268]: I have copied yarn.lock, from main and run `yarn`

File: packages/cache-service/src/data-packages/data-packages.service.ts [hunk 8]
```diff
@@ -23,51 +35,87 @@ import {
...
-      savePromises.push(saveToBundlrPromise);
-    }
 
     await Promise.allSettled(savePromises);
   }
 
-  // eslint-disable-next-line @typescript-eslint/class-methods-use-this
-  async saveManyDataPackagesInDB(dataPackages: CachedDataPackage[]) {
-    await DataPackage.insertMany(dataPackages);
+  private performBroadcast(
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/cache-service/src/data-packages/data-packages.service.ts:98]: For better stack traces?

File: packages/cache-service/test/e2e/data-packages.spec.ts [hunk 9]
```diff
@@ -38,9 +38,11 @@ import {
   mockSigner,
   produceMockDataPackage,
 } from "../common/mock-values";
-import { connectToTestDB, dropTestDatabase } from "../common/test-db";
+import {
+  createTestDB as createAndConnectToTestDb,
```
Comment (rpiszczatowski) [inline packages/cache-service/test/e2e/data-packages.spec.ts:42]: Maybe you can simply change the original method name :)

Comment (koonopek) [reply packages/cache-service/test/e2e/data-packages.spec.ts:42]: ohh, that is how ts refactor works in vscode, will fix in new PR

**Review by hatskier — APPROVED** [review]

**Review by cehali — APPROVED** [review]
👍

**Pull request merged**